src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCacheEntry.cs

```diff
@@ -1,11 +1,81 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
+using UnityEngine;
+
 namespace Microsoft.MixedReality.SpectatorView
 {
+    [Serializable]
+    internal class AssetId
+    {
+        public static AssetId Empty { get; } = new AssetId(System.Guid.Empty, -1);
+
+        /// <summary>
+        /// The Unity guid for the Asset
+        /// </summary>
+        public StringGuid Guid => guid;
+
+        [SerializeField]
+        private StringGuid guid;
+
+        /// <summary>
+        /// The Unity file identifier for the Asset
+        /// </summary>
+        public long FileIdentifier => fileIdentifier;
+
+        [SerializeField]
+        private long fileIdentifier;
+
+        public AssetId(StringGuid guid, long fileIdentifier)
+        {
+            this.guid = guid;
+            this.fileIdentifier = fileIdentifier;
+        }
+
+        public override bool Equals(object obj)
+        {
+            AssetId assetId = obj as AssetId;
+            if (assetId == null)
+            {
+                return false;
+            }
+
+            return this == assetId;
+        }
+
+        public override int GetHashCode()
+        {
+            return FileIdentifier.GetHashCode() ^ Guid.ToString().GetHashCode();

```

Comment:
This will do an allocation everytime you add to the dictionary, because GetHashCode is called, which will create a string from Guid.

Instead, just do Guid.GetHashCode().

Reply:
Sounds good. I'll pick these changes up in stabilization/1.1.0 with the expectation to merge back into master